kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -155,7 +158,7 @@ public String toString() {
                 + ", authentication=" + authentication
                 + ", sniHostname='" + sniHostname + '\''
                 + ", pendingReadComplete=" + pendingReadComplete
-                + ", isClientBlocked=" + isClientBlocked
+                + ", blocked=" + progressionLatch
```

Comment:
Not sure logging `blocked=2` is helpful to most consumers. I do wonder if an enum might be more helpful.

Reply:
Fair. We have never made the time or found the motivation to properly pull the frontend handler into the channel state machine. So yes there are probably a few issues still lurking around here. 

Happy to leave this as is rather than poking that hornets nest right now.